### DISCUSSION TITLE
Provide thorough test data

### DISCUSSION THREAD

**Discussion by jotaen** [jotaen/xit#63] [Other]
For tools or editor plugins it would be useful to have a thorough test file, to check whether all rules and edge-cases are handled properly.

I’ve already shared something in https://github.com/jotaen/xit/discussions/6, but that could be more thorough and better structured.

It’s probably best to provide a copy-able file on the website, and then have separate annotations for more details.

**Comment by jotaen** [comment]
I’ve turned the test file (which I used internally for checking the Sublime Plugin) into something proper here: https://xit.jotaen.net/syntax-guide

It’s not an automated test, but it should already be helpful both for users who want to learn more about the rules, and for tool creators to compare their implementation against it, or derive test cases.

There are probably some edge-cases missing, so I’ll just update that document whenever something comes up.

If you’d like something to be added (or corrected), please post it into this issue here as comment.

**Comment by jotaen** [comment]
One additional remark, by the way: especially for syntax highlighters, it’s probably impossible to get **all** of the rules correct, due to various limitations in the highlighting engines. I think that’s okay, though, since a highlighter should be considered to be a visual aid, rather than an authoritative verification. Marking things as invalid should also be more of a nice-to-have.

**Comment by vcsala** [comment]
I used sematic highlighting in my latest version of my vscode plugin (xit! extended). It loads a bit slower but it follows the specification very closely (for example, only highlighting the first due date). I am considering to use this to implement different highlights for overdue dates.

**Comment by jotaen** [comment]
> I used sematic highlighting in my latest version of my vscode plugin (xit! extended). It loads a bit slower but it follows the specification very closely (for example, only highlighting the first due date). I am considering to use this to implement different highlights for overdue dates.

👍  I think VSCode has one of the more powerful highlighting engines, so it’s cool that such enhanced features are possible there.
